### PR TITLE
Slow test fixup

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description
Increase reliability of capture proxy tests by correctly passing longTest tag and lowering concurrency to numberOfCores/2
* Category: Test Fix
* Why these changes are required? Currently, proxy tests are running for both slow test 
* What is the old behavior before changes and new behavior after changes? Current, KafkaConfigurationCaptureProxyTest tests run for both test and slowTest tasks which is not desired, new behavior is that KafkaConfigurationCaptureProxyTest only runs with SlowTest with dynamic parallelization in half.

### Issues Resolved
N/A

Is this a backport? If so, please add backport PR # and/or commits #
N/A

### Testing
Tested with gradle test to ensure tests are not being run

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/testcontainers/annotations/KafkaContainerTest.java [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
 import org.junit.jupiter.api.parallel.ResourceLock;
 
 @Inherited
 @ResourceLock("KafkaContainer")
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
```
Comment (gregschohn) [inline TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/testcontainers/annotations/KafkaContainerTest.java:13]: What are these lines doing here?  Why do you need them?

**Review by gregschohn — COMMENTED** [review]
How are you limiting these tests to NOT run for the test target?

File: TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/testcontainers/annotations/KafkaContainerTest.java (hunk 1 above)
Comment (AndreKurait) [reply TrafficCapture/trafficCaptureProxyServer/src/test/java/org/opensearch/migrations/trafficcapture/proxyserver/testcontainers/annotations/KafkaContainerTest.java:13]: Java by default only adds annotations to the .class file and are not loaded at runtime, for Junit Tags to work, these annotations need to be included at runtime.

**Comment by AndreKurait** [comment]
> How are you limiting these tests to NOT run for the test target?

The inner `@TestContainerTest` defines the @Tag("longTest") which is excluded from the test task.

**Review by gregschohn — APPROVED** [review]

**Pull request merged**